### PR TITLE
Don't release already-created intermediate *Value

### PR DESCRIPTION
**Pull request by jeromegn**
I started getting issues where values I had pre-created and saved for later use were getting released when using `ctx.Create`. When my code called ctx.Create a second time with the previously-saved `*Value`s, it would segfault due to a null pointer.

This PR checks if the map value, slice value and struct value types are of `*Value` and will not release them automatically if that's the case.

I'm sure there are prettier ways of doing this, but it seems the simplest without changing function signatures.

### CODE REVIEW

File: v8_create.go [hunk 1]
```diff
@@ -236,14 +248,20 @@ func (ctx *Context) writeStructFields(ob *Value, val reflect.Value) error {
 		}
 
 		v8Tags := strings.Split(f.Tag.Get("v8"), ",")
-		v, err := ctx.createWithTags(val.Field(i), v8Tags)
+		toCreate := val.Field(i)
+		// if toCreate.Kind() == reflect.Interface {
+		// 	toCreate = toCreate.Elem()
+		// }
```
Comment (augustoroman) [inline v8_create.go]: remove dead code

File: v8_create.go [hunk 2]
```diff
@@ -145,14 +145,20 @@ func (ctx *Context) createWithTags(val reflect.Value, tags []string) (*Value, er
 		keys := val.MapKeys()
 		sort.Sort(stringKeys(keys))
 		for _, key := range keys {
-			v, err := ctx.create(val.MapIndex(key))
+			toCreate := val.MapIndex(key)
+			if toCreate.Kind() == reflect.Interface {
+				toCreate = toCreate.Elem()
```
Comment (augustoroman) [inline v8_create.go]: The tests are great for verifying that pre-existing valuePtrTypes are released, but it doesn't check for this interface de-referencing.  I think it's safer to have `.create` return a boolean that indicates whether a value was allocated.

**Comment by augustoroman** [comment]
Thanks for catching this!

Addressed by https://github.com/augustoroman/v8/commit/13162a9353a2171eb720332006af4aaf1ddf626a

**Comment by jeromegn** [comment]
Thanks!

**Pull request closed without merging**